plugins/inputs/sqlserver/azuresqlqueries.go

```diff
@@ -1196,3 +1320,261 @@ SELECT
 	,DATABASEPROPERTYEX(DB_NAME(), 'Updateability') as replica_updateability
 FROM sys.dm_os_schedulers AS s
 `
+
+const sqlAzureMIPartQueryPeriod = `
+DECLARE @QueryData NVARCHAR(MAX) = ?1; --input parameter
+
+IF SERVERPROPERTY('EngineEdition') <> 8 BEGIN /*not Azure Managed Instance*/
+	DECLARE @ErrorMessage AS nvarchar(500) = 'Telegraf - Connection string Server:'+ @@SERVERNAME + ',Database:' + DB_NAME() +' is not an Azure Managed Instance. Check the database_type parameter in the telegraf configuration.';
+	RAISERROR (@ErrorMessage,11,1)
+	RETURN
+END;
+
+IF OBJECT_ID ('tempdb.dbo.#ExecForeachDb') IS NOT NULL DROP PROCEDURE #ExecForeachDb;
+
+/* Create a temporary procedure that iterates through all DBs that the user has access to in this managed instance, except system DBs, and execute the query specified with @queryText */
+EXEC('CREATE PROCEDURE #ExecForeachDb @queryText NVARCHAR(MAX)
+AS
+BEGIN
+	SET NOCOUNT ON;
+
+	DECLARE dbCursor CURSOR LOCAL FOR 
+	SELECT 
+		[name] 
+	FROM sys.databases
+	WHERE HAS_DBACCESS([name]) = 1 
+		AND [name] NOT IN (''master'', ''tempdb'', ''model'', ''msdb'');
+	DECLARE @CurrSqlText nvarchar(max);
+	DECLARE @dbName sysname;
+
+	OPEN dbCursor;
+	FETCH NEXT FROM dbCursor INTO @dbName;
+
+	WHILE @@Fetch_Status=0 BEGIN
+		SET @CurrSqlText = N''USE ''+ QUOTENAME(@dbName ,''"'') + @queryText
```

Comment:
A database may be dropped between the time the cursor is open and the time this USE executes.